src/ripple/shamap/SHAMapMissingNode.h

```diff
@@ -38,28 +39,37 @@ class SHAMapMissingNode
 {
 private:
     SHAMapType mType;
-    uint256 mNodeHash;
+    SHAMapHash mNodeHash;
+    uint256    mNodeID;
 public:
     SHAMapMissingNode (SHAMapType t,
-                       uint256 const& nodeHash)
+                       SHAMapHash const& nodeHash)
         : std::runtime_error ("SHAMapMissingNode")
         , mType (t)
         , mNodeHash (nodeHash)
     {
     }
 
+    SHAMapMissingNode (SHAMapType t,
+                       uint256 const& nodeID)
+        : std::runtime_error ("SHAMapMissingNode")
+        , mType (t)
+        , mNodeID (nodeID)
+    {
+    }
+
     SHAMapType getMapType () const
     {
         return mType;
     }
 
-    uint256 const& getNodeHash () const
+    SHAMapHash const& getNodeHash () const
```

Comment:
I suggest removing this `getNodeHash()` method.  It seems to be unused at the moment (I commented it out and successfully built).  If someone were to call it they might be surprised it they got a `zero` back because the `SHAMapMissingNode` was constructed with a `uint256`.

You could also remove the `getMapType()` method if you adjust the streaming operator (which is already a friend).  But all that will do is improve our code coverage :smile:.